Sources/SwiftGen/Commander/XCFileListCommands.swift

```diff
@@ -0,0 +1,94 @@
+//
+// SwiftGen
+// Copyright © 2020 SwiftGen
+// MIT Licence
+//
+
+import Commander
+import PathKit
+import StencilSwiftKit
+import SwiftGenKit
+
+enum XCFileListCLI {
+  private enum CLIOption {
+    static let config = Option<Path>(
+      "config",
+      default: "swiftgen.yml",
+      flag: "c",
+      description: "Path to the configuration file to use",
+      validator: checkPath(type: "config file") { $0.isFile }
+    )
+
+    static let inputs = Option<Path>(
+      "inputs",
+      default: "./swiftgen-input.xcfilelist",
+      flag: "i",
+      description: "Path to write the xcfilelist for the input files"
+    )
+
+    static let outputs = Option<Path>(
+      "outputs",
+      default: "./swiftgen-output.xcfilelist",
+      flag: "o",
+      description: "Path to write the xcfilelist for the output files"
+    )
+  }
+
+  static let generate = command(
+    CLIOption.config,
+    CLIOption.inputs,
+    CLIOption.outputs
+  ) { configPath, inputPath, outputPath in
+    try ErrorPrettifier.execute {
+      let config = try Config(file: configPath)
+
+      try writeInputsXCFileList(for: config, to: inputPath)
+      try writeOutputsXCFileList(for: config, to: outputPath)
+    }
+  }
+
+  private static func writeInputsXCFileList(for config: Config, to path: Path) throws {
+    let inputPaths = pathsByCommand(for: config) { entry -> [Path] in
+      entry.inputs.map { (config.sourcePath + $0).absolute() }
```

Comment:
This seems like a bigger change and I might not have time to do it until later in the week. 

I did take a quick look at it and `Parser` doesn't feel like the right place for it (but I. am new to the codebase). I was thinking maybe `ConfigEntry` would make sense since it already contains the `inputs` and `filter`. Maybe there could be a `filteredInputPaths` variable that returns an array of matching paths. How does that sound?

Reply:
So, I've checked the codebase. The logic for searching & parsing is already in the `Parser` protocol:
https://github.com/SwiftGen/SwiftGen/blob/stable/Sources/SwiftGenKit/Parsers/Parser.swift#L57-L70

We can copy (or extract) the logic for recursing & matching a filter, into a separate property (or function) so we can use it here.